packages/api/src/controllers/stream.ts

```diff
@@ -902,6 +902,7 @@ app.post(
     const patch: Partial<DBSession & DBStream> = {
       isHealthy: payload.is_active ? payload.is_healthy : undefined,
       issues,
+      isActive: payload.is_active,
```

Comment:
As per the comment on L895 above, this can introduce other race conditions with recordings instead, which is the case if the health hook arrives before than the `/setactive` hook, with the `is_active` being `false`. This will cause the recordings not to be generated for that stream because the `/setactive` will detect the change is a noop here https://github.com/livepeer/livepeer-com/blob/2d1e24eaf8ef9eaa089936bea47373d696f85a93/packages/api/src/controllers/stream.ts#L1120

@gioelecerati had previously fixed this on https://github.com/livepeer/studio/pull/1916, by replacing the "read and replace" with a single "patch" of the stream object. It seems to have been reverted later for an unclear reason ("until we're sure it's not causing issues") on https://github.com/livepeer/studio/pull/1954, but I believe it is the better fix for this not to introduce other potential race conditions, which I believe this version does.

@gioelecerati was there any conclusion about the other fix introducing any issues?